awx/main/tests/unit/test_validators.py

```diff
@@ -12,12 +12,43 @@
     TEST_OPENSSH_KEY_DATA,
     TEST_OPENSSH_KEY_DATA_LOCKED,
     TEST_SSH_CERT_KEY,
+    TEST_CATTED_SSH_KEY_DATA,
 )
 from rest_framework.serializers import ValidationError as RestValidationError
 
 import pytest
 
 
+def test_invalid_keys():
+    invalid_keys = [
+        "---BEGIN FOO -----foobar-----END FOO----",
+        "-----BEGIN FOO---foobar-----END FOO----",
+        "-----BEGIN FOO-----foobar---END FOO----",
+        "-----  BEGIN FOO  ----- foobar -----  FAIL FOO  ----",
+        "-----  FAIL FOO ----- foobar -----  END FOO  ----",
+        "----BEGIN FOO----foobar----END FOO--------BEGIN FOO----foobar----END FOO----"
```

Comment:
This is a corner-case that I know is a valid pair of keys but the regular expression with `-{4,}` will gather up all the way to the second `BEGIN` and cause a failure.